File: sixthspace/src/main/java/com/jyh/sixthspace/ui/activity/MainActivity.java
package com.jyh.sixthspace.ui.activity;

import android.os.Bundle;


import android.view.MenuItem;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.viewpager.widget.ViewPager;

import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.jyh.sixthspace.R;
import com.jyh.sixthspace.databinding.ActivityMainBinding;
import com.jyh.sixthspace.ui.adapter.MainPagerAdapter;
import com.jyh.sixthspace.uilibrary.StatusBarUtil;


public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding binding;
    private BottomNavigationView navigation;
    private MainPagerAdapter mainPagerAdapter;
    private int mAlpha = StatusBarUtil.DEFAULT_STATUS_BAR_ALPHA;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        setStatusBar();
        initView();
    }

    private void initView() {
        navigation = binding.navigation;
        navigation.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener);
        mainPagerAdapter = new MainPagerAdapter(getSupportFragmentManager());
        binding.vpMain.setAdapter(mainPagerAdapter);
        binding.vpMain.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

            }

            @Override
            public void onPageSelected(int position) {
                setStatusBar();
                switch (position) {

                    case 0:
                        navigation.setSelectedItemId(R.id.navigation_recommend);
                        break;
                    case 1:
                        navigation.setSelectedItemId(R.id.navigation_movies);
                        break;
                    case 2:
                        navigation.setSelectedItemId(R.id.navigation_live);
                        break;
                }

            }

            @Override
            public void onPageScrollStateChanged(int state) {

            }
        });

    }

    private BottomNavigationView.OnNavigationItemSelectedListener mOnNavigationItemSelectedListener = new BottomNavigationView.OnNavigationItemSelectedListener() {
        @Override
        public boolean onNavigationItemSelected(@NonNull MenuItem item) {
            setStatusBar();
            switch (item.getItemId()) {
                case R.id.navigation_recommend:
                    binding.vpMain.setCurrentItem(0);
                    return true;
                case R.id.navigation_movies:
                    binding.vpMain.setCurrentItem(1);
                    return true;
                case R.id.navigation_live:
                    binding.vpMain.setCurrentItem(2);
                    return true;
            }
            return false;
        }
    };


    protected void setStatusBar() {
//        mStatusBarColor = getResources().getColor(R.color.colorPrimary);
//        StatusBarUtil.setColorForDrawerLayout(this, binding.drawerLayout, mStatusBarColor, mAlpha);
        StatusBarUtil.setTranslucentForImageViewInFragment(this, null);
    }
}

File: app/src/main/java/com/sixth/space/ui/MainActivity.kt
package com.sixth.space.ui

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.activity.viewModels
import androidx.core.splashscreen.SplashScreen.Companion.installSplashScreen
import com.sixth.space.databinding.ActivityMainBinding
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.sixth.space.model.MainViewModel
import com.sixth.space.R

class MainActivity : AppCompatActivity() {

    private val viewModel by viewModels<MainViewModel>()
    private lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        installSplashScreen().apply {
            setKeepOnScreenCondition {
                !viewModel.go2main.value
            }
        }
        initView();

    }
    private fun initView() {
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        binding.viewPager2.adapter = MainFragmentStateAdapter(this);
        binding.bottomNavigationView.setOnItemSelectedListener {
            when (it.itemId) {
                R.id.navigation_home -> {
                    binding.viewPager2.setCurrentItem(0, true);
                }

                R.id.navigation_discovery -> {
                    binding.viewPager2.setCurrentItem(1, true);
                }

                R.id.navigation_hot -> {
                    binding.viewPager2.setCurrentItem(2, true);
                }
            }
            true
        }
    }
}

class MainFragmentStateAdapter(fragmentActivity: FragmentActivity) :
    FragmentStateAdapter(fragmentActivity) {
    override fun getItemCount(): Int = 3
    override fun createFragment(position: Int): Fragment {
        return when (position) {
            0 -> HomeFragment()
            1 -> DiscoveryFragment()
            else -> HotFragment();
        }
    }

}
